File: csharp/1423.maximum-points-you-can-obtain-from-cards.cs
/*
 * @lc app=leetcode id=1423 lang=csharp
 *
 * [1423] Maximum Points You Can Obtain from Cards
 */

using System;
using System.Linq;

// @lc code=start
public class Solution {
    public int MaxScore(int[] cardPoints, int k) {
        int acc = cardPoints.Take(k).Sum(), ret = acc, n = cardPoints.Length;
        for (int i = 1; i <= k; ++i) {
            acc = acc - cardPoints[k - i] + cardPoints[n - i];
            ret = Math.Max(ret, acc);
        }
        return ret;
    }
}
// @lc code=end

File: java/1423.maximum-points-you-can-obtain-from-cards.java
/*
 * @lc app=leetcode id=1423 lang=java
 *
 * [1423] Maximum Points You Can Obtain from Cards
 */

// @lc code=start
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int n = cardPoints.length, acc = 0, ret;
        for (int i = 0; i < k; ++i) {
            acc += cardPoints[i];
        }
        ret = acc;
        for (int i = 0; i < k; ++i) {
            acc += cardPoints[n - 1 - i] - cardPoints[k - 1 - i];
            ret = Math.max(ret, acc);
        }
        return ret;
    }
}
// @lc code=end

